include/zephyr/sdll/sdll.h

```diff
@@ -0,0 +1,228 @@
+ *
+ * @param cid SDLL context id
+ * @param data Pointer to the received data
+ * @param len Length of the received data
+ */
+typedef void (*sdll_frame_received_callback_t)(const sdll_context_id cid, const uint8_t * data, const size_t len);
+
+/**
+ * @brief SDLL frame validation function
+ *
+ * This function is called after a frame is completely received and before
+ * calling `frame_received()`. It is used to validate the content of the frame
+ * (i.e. a checksum or CRC-16 check).
+ *
+ * @param data Pointer to the received data
+ * @param len Length of the received data
+ * @return true if the frame is valid, false otherwise
+  */
+typedef bool (*sdll_frame_validation_check_function_t)(const sdll_context_id cid, const uint8_t * data, const
+size_t len);
+
+/**
+ * @brief SDLL frame send function
+ *
+ * This function is called to send a frame.
+ *
+ * @param data Pointer to the data to send
+ * @param len Length of the data to send
+ * @return Number of bytes sent
+ * @return -EINVAL if one or more parameters are invalid
+ */
+typedef int (*sdll_frame_send_function_t) (const sdll_context_id cid, const uint8_t * data,
+    const size_t len);
+
+/**
+ * @brief SDLL frame sent callback
+ *
+ * This function is called when a frame is completely sent.
+ *
+ * @param data Pointer to the sent data
+ * @param len Length of the sent data
+ */
+typedef void (*sdll_frame_sent_callback_t) (const sdll_context_id cid, const uint8_t * data, const size_t len);
+
+/**
+ * @brief SDLL configuration
+ *
+ * This structure is used to configure the SDLL frame receiver
+ * on initialization.
+ */
+struct sdll_receiver_config {
+
+    /** @brief Buffer to store frames received */
+    uint8_t * receive_buffer;
+
+    /** @brief Length of the buffer. Must be at least
+     * `SDLL_MINIMUM_BUFFER_SIZE` bytes */
+    size_t receive_buffer_len;
+
+    /** @brief Frame received callback */
+    sdll_frame_received_callback_t frame_received_cb;
+
+    /** @brief Frame check function */
+    sdll_frame_validation_check_function_t frame_check_fn;
+};
+
+/**
+ * @brief SDLL configuration
+ *
+ * This structure is used to configure the SDLL frame transmitter
+ * on initialization.
+ */
+struct sdll_transmitter_config {
+
+    /** @brief Buffer to store frames to be sent */
+    uint8_t * send_buffer;
+
+    /** @brief Length of the buffer. Must be at least
+     * `SDLL_MINIMUM_BUFFER_SIZE` bytes */
+    size_t send_buffer_len;
+
+    /** @brief Frame send function */
+    sdll_frame_send_function_t frame_send_fn;
+
+#ifdef CONFIG_SDLL_ASYNC
+    /** @brief Frame sent callback */
+    sdll_frame_sent_callback_t frame_sent_cb;
+#endif /* CONFIG_SDLL_ASYNC */
+};
+
+/**
+ * @brief SDLL initialization function
+ *
+ * This function initializes the SDLL library and configures both the transmitter
+ * stage and/or the receiver stage if provided.
+ *
+ * @note:
+ * The buffer size must be at least `SDLL_MINIMUM_BUFFER_SIZE` bytes.
```

Comment:
```suggestion
 * @pre: The buffer size must be at least `SDLL_MINIMUM_BUFFER_SIZE` bytes.
```
so we mark it as a pre-condition.